flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -978,6 +998,18 @@ private void shutdownAsyncThreads() throws Exception {
         }
     }
 
+    private static Closeable toCloseable(AutoCloseable autoCloseable) {
+        return () -> {
+            try {
+                autoCloseable.close();
+            } catch (IOException ex) {
+                throw ex;
+            } catch (Exception exception) {
+                rethrow(exception);
+            }
+        };
+    }
+
```

Comment:
Just a quick sum-up on this based on the offline discussion:

it is better not to wrap to RuntimeException because it changes the footprint of StreamTask

Reply:
I have updated the PR with new implementation